### PR TITLE
[spi_device/testplan] Added TPM specific testpoints

### PR DESCRIPTION
**Pull request by jason-hoddinett**
merged similar tests & added covergroups.

Signed-off-by: Jason Hoddinett <jason.hoddinett.ensilica@opentitan.org>

### CODE REVIEW

**Comment by jason-hoddinett** [comment]
@weicaiyang @eunchan @sriyerg

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 1]
```diff
@@ -146,5 +146,102 @@
...
+      name: tpm_fifo_read
+      desc: '''
+            - Set TPM_CFG.TPM_MODE to 0.
+            - Set TPM_CFG.EN.
+            - Randomise other fields in TPM_CFG.
+            - Send TPM read command over the SPI bus with a randomised address.
+            - Check TPM_CMD_ADDR.
+            - Check TPM_STATUS.cmdaddr_notempty and INTR_STATE.tpm_header_notempty, they should be asserted if hw_reg_dis == 0.
+            - If hw_reg_dis == 0, the data is returned to the host via return-by-HW register, else the data is returned via read FIFO.
+            - Check TPM_CFG.INVALID_LOCALITY.
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:159]: this should be randomized, right? I will affect if the command and address are updated to FIFO or the data directly returns by HW.

**Review by weicaiyang — COMMENTED** [review]
seems like we still haven't captured how to verify `activeLocality`. @eunchan can you help to clarify it, so that Jason can add a test to cover that.

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 2]
```diff
@@ -146,5 +146,99 @@
       milestone: V2
       tests: []
     }
+    {
+      name: tpm_read
+      desc: '''
+            - Set TPM_CFG.TPM_MODE to 0 and set TPM_CFG.EN.
+            - Randomise other fields in TPM_CFG.
+            - Assert the tpm_csb.
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:154]: nit: csb is active low. should be de-assert

File: hw/ip/spi_device/data/spi_device_testplan.hjson [hunk 3]
```diff
@@ -146,5 +146,99 @@
...
+            - Compare this value with the expected value.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_write
+      desc: '''
+            - Set TPM_CFG.TPM_MODE to 0 and set TPM_CFG.EN.
+            - Randomise other fields in TPM_CFG.
+            - Assert the tpm_csb.
```
Comment (weicaiyang) [inline hw/ip/spi_device/data/spi_device_testplan.hjson:172]: ditto

**Review by weicaiyang — APPROVED** [review]
LGTM with a nit

@eunchan can you also take a look?

File: hw/ip/spi_device/data/spi_device_testplan.hjson (hunk 2 above)
Comment (jason-hoddinett) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:154]: 'assert' is agnostic to signal level. Asserting an active low is equivalent to driving low.

Comment (jason-hoddinett) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:154]: Would you prefer a more explicit description, eg "drive the csb low"

Comment (weicaiyang) [reply hw/ip/spi_device/data/spi_device_testplan.hjson:154]: yeah, that also makes sense. don't need to change it.

**Review by eunchan — COMMENTED** [review]
LGTM

**Pull request merged**